pyrogram/methods/messages/copy_media_group.py

```diff
@@ -0,0 +1,134 @@
+            from_chat_id (``int`` | ``str``):
+                Unique identifier (int) or username (str) of the source chat where the original media group was sent.
+                For your personal cloud (Saved Messages) you can simply use "me" or "self".
+                For a contact that exists in your Telegram address book you can use his phone number (str).
+
+            message_id (``int``):
+                Message identifier in the chat specified in *from_chat_id*.
+
+            captions (``string`` or list of ``string`` , *optional*):
+                New caption for media, 0-1024 characters after entities parsing for each media.
+                If not specified, the original caption is kept.
+                Pass "" (empty string) to remove the caption.
+
+                If a ``string`` is passed, it becomes a caption only for the first media.
+                If a list of ``string`` passed, each element becomes caption for each media element.
+                You can pass ``None`` in list to keep the original caption. Please check examples.
+
+            disable_notification (``bool``, *optional*):
+                Sends the message silently.
+                Users will receive a notification with no sound.
+
+            reply_to_message_id (``int``, *optional*):
+                If the message is a reply, ID of the original message.
+
+            schedule_date (``int``, *optional*):
+                Date when the message will be automatically sent. Unix time.
+
+        Returns:
+            List of :obj:`~pyrogram.types.Message`: On success, a list of copied messages is returned.
+
+        Example:
+            .. code-block:: python
+
+                # Copy a media group
+                app.copy_media_group("me", "pyrogramlounge", 470061)
+                app.copy_media_group("me", "pyrogramlounge", 470061, captions="Single caption")
+                app.copy_media_group("me", "pyrogramlounge", 470061, captions=["Media 1 caption", None, ""])
+        """
+
+        media_group = await self.get_media_group(from_chat_id, message_id)
+        multi_media = []
+
+        for i, message in enumerate(media_group):
+
+            if message.photo:
+                file_id = message.photo.file_id
+            elif message.audio:
+                file_id = message.audio.file_id
+            elif message.document:
+                file_id = message.document.file_id
+            elif message.video:
+                file_id = message.video.file_id
+            else:
+                raise TypeError("Message with this type can't be copied.")
+
+            media = utils.get_input_media_from_file_id(file_id=file_id)
+            multi_media.append(
+                raw.types.InputSingleMedia(
+                    media=media,
+                    random_id=self.rnd_id(),
+                    **await self.parser.parse(
+                        captions[i] if type(captions) is list and i < len(captions) and captions[i] is not None else
```

Comment:
`type(captions) is list` can replaced with  `isinstance(captions, list)`

Reply:
Well seems like a nice idea